src/SortChecker.cpp

```diff
@@ -114,19 +114,32 @@ class Visitor : public RecursiveASTVisitor<Visitor> {
   }
 #endif
 
-  typedef enum {
+  enum CompareFunction {
     CMP_FUNC_UNKNOWN = 0,
     CMP_FUNC_SORT,
     CMP_FUNC_BINARY_SEARCH,
+    CMP_FUNC_LOWER_BOUND,
     // TODO: other APIs from
     // https://en.cppreference.com/w/cpp/named_req/Compare
     CMP_FUNC_NUM
-  } CompareFunction;
+  };
+
+  bool isKindOfBinarySearch(CompareFunction func) {
+    switch (func) {
+    case CMP_FUNC_BINARY_SEARCH:
+      LLVM_FALLTHROUGH;
+    case CMP_FUNC_LOWER_BOUND:
```

Comment:
Why not just
```
    case CMP_FUNC_BINARY_SEARCH:
    case CMP_FUNC_LOWER_BOUND:
```
? Fallthrough annotations are normally omitted in such cases.

Reply:
Well, didn't fallthrough annotations added for such cases? 
For example, clang source code:
https://github.com/llvm/llvm-project/blob/be7e938e2712f3d84d1775ecdb1bc6784ea114f7/clang/lib/Interpreter/IncrementalParser.cpp#L68-L73

Reply:
There instances of another style too: https://github.com/llvm/llvm-project/blob/be7e938e2712f3d84d1775ecdb1bc6784ea114f7/clang/lib/CodeGen/CGException.cpp#L154 and AFAIK it's much more common, both in LLVM and beyond...

Reply:
Ok, now I get what you mean
Will fix